src/main/kotlin/me/serce/solidity/ide/run/SolidityRunConfig.kt

```diff
@@ -0,0 +1,199 @@
+import org.jdom.Element
+import java.util.*
+
+
+class SolidityRunConfig(configurationModule: SolidityRunConfigModule?, factory: ConfigurationFactory?) : ModuleBasedConfiguration<SolidityRunConfigModule>(configurationModule, factory), CommonJavaRunConfigurationParameters {
+
+  var myData: Data = Data()
+
+  override fun getEnvs(): Map<String, String> {
+    return myData.envs;
+  }
+
+  override fun setAlternativeJrePath(p0: String?) {
+    myData.ajre = p0
+  }
+
+  override fun isPassParentEnvs(): Boolean {
+    return true
+  }
+
+  override fun setProgramParameters(p0: String?) {
+    myData.programParameters = p0
+  }
+
+  override fun setVMParameters(p0: String) {
+    myData.vmParameters = p0
+  }
+
+  override fun isAlternativeJrePathEnabled(): Boolean {
+    return myData.ajreEnabled
+  }
+
+  override fun getPackage(): String? {
+    throw UnsupportedOperationException("Should not be called for this class")
+  }
+
+  override fun getRunClass(): String? {
+    throw UnsupportedOperationException("Should not be called for this class")
+  }
+
+  override fun getWorkingDirectory(): String? {
+    return myData.getWorkingDirectory()
+  }
+
+  override fun setAlternativeJrePathEnabled(p0: Boolean) {
+    myData.ajreEnabled = p0
+  }
+
+  override fun getVMParameters(): String? {
+    return myData.vmParameters
+  }
+
+  override fun setWorkingDirectory(p0: String?) {
+    myData.setWorkingDirectory(p0)
+  }
+
+  override fun setEnvs(p0: MutableMap<String, String>) {
+    myData.envs.clear()
+    myData.envs.putAll(p0)
+  }
+
+  override fun setPassParentEnvs(p0: Boolean) {
+  }
+
+  override fun getProgramParameters(): String? {
+    return myData.programParameters;
+  }
+
+  override fun getAlternativeJrePath(): String? {
+    return myData.ajre
+  }
+
+  override fun getValidModules(): Collection<Module> {
+    return ModuleManager.getInstance(project).modules.asList()
+  }
+
+  override fun getState(p0: Executor, env: ExecutionEnvironment): RunProfileState? {
+    val state = SolidityRunState( env, this)
+    state.consoleBuilder = TextConsoleBuilderFactory.getInstance().createBuilder(project)
+    return state
+  }
+
+  override fun getConfigurationEditor(): SettingsEditor<out RunConfiguration> {
+    val group = SettingsEditorGroup<SolidityRunConfig>()
+    group.addEditor(ExecutionBundle.message("run.configuration.configuration.tab.title", *arrayOfNulls(0)), SolidityConfigurableEditorPanel(this.project))
+    group.addEditor(ExecutionBundle.message("logs.tab.title", *arrayOfNulls(0)), LogConfigurationPanel())
+    return group
+  }
+
+
+  @Suppress("DEPRECATION")
+  override fun readExternal(element: Element?) {
+    super.readExternal(element)
+    JavaRunConfigurationExtensionManager.getInstance().readExternal(this, element!!)
+    com.intellij.openapi.util.DefaultJDOMExternalizer.readExternal(this, element)
+    com.intellij.openapi.util.DefaultJDOMExternalizer.readExternal(myData, element)
```

Comment:
Was there any problem with `com.intellij.util.xmlb.XmlSerializer`? There is a change that `DefaultJDOMExternalizer` will be deleted in future versions of openapi.

Reply:
There was a problem when I was trying the first time. But now it seems it was just a bug in my code.
Done